config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
   # Defines the root path route ("/")
   # root "articles#index"
+
+  get '/users', to: 'users#index', as: 'all_users'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/posts/:id', to: 'posts#show', as: 'post'
```

Comment:
- [ Optional ] Good job setting all the routes as stated in the project requirements, however, at times typing the routes as you've done might lead to a route being mis typed. I would like to suggest that you use the rails `resources` feature which generates the routes and matches them to the respective `method` in the controllers. Please see the example below for more details on this.

## Example code
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```